lib/pipelines.js

```diff
@@ -1,259 +1,294 @@
+  query,
+  fs.readFileSync('./sql/delete_pipeline_couplings_by_pipeline.sql').toString('utf8'), (d) => d,
+);
 async function http_delete(pg_pool, req, res, regex) {
-  let pipeline_key = httph.first_match(req.url, regex)
-  let pipeline = await select_pipeline(pg_pool, [pipeline_key.toLowerCase()])
-  if(pipeline.length === 0) {
-    throw new common.NotFoundError('The specified pipeline was not found.')
+  const pipeline_key = httph.first_match(req.url, regex);
+  const pipeline = await select_pipeline(pg_pool, [pipeline_key.toLowerCase()]);
+  if (pipeline.length === 0) {
+    throw new common.NotFoundError('The specified pipeline was not found.');
   }
   // Remove the app couplings
-  await delete_pipeline_couplings_by_pipeline_query(pg_pool, [pipeline[0].pipeline])
+  await delete_pipeline_couplings_by_pipeline_query(pg_pool, [pipeline[0].pipeline]);
   // Remove the pipeline
-  await delete_pipeline_query(pg_pool, [pipeline[0].pipeline])
-  httph.ok_response(res, 
-    JSON.stringify(pipeline_postgres_to_response(pipeline[0])))
+  await delete_pipeline_query(pg_pool, [pipeline[0].pipeline]);
+  httph.ok_response(res,
+    JSON.stringify(pipeline_postgres_to_response(pipeline[0])));
 }
 
-const select_pipeline_coupling_by_app = query.bind(query, fs.readFileSync('./sql/select_pipeline_coupling_by_app.sql').toString('utf8'), (d) => { return d; });
+const select_pipeline_coupling_by_app = query.bind(
+  query,
+  fs.readFileSync('./sql/select_pipeline_coupling_by_app.sql').toString('utf8'), (d) => d,
+);
 async function http_get_pipeline_coupling_by_app(pg_pool, req, res, regex) {
-  let app_key = httph.first_match(req.url, regex)
-  let app = await common.app_exists(pg_pool, app_key)
-  let pipeline_coupling = await select_pipeline_coupling_by_app(pg_pool, [app.app_uuid])
-  if(pipeline_coupling.length === 0) {
-    throw new common.NotFoundError('The specified pipeline coupling was not found.')
+  const app_key = httph.first_match(req.url, regex);
+  const app = await common.app_exists(pg_pool, app_key);
+  const pipeline_coupling = await select_pipeline_coupling_by_app(pg_pool, [app.app_uuid]);
+  if (pipeline_coupling.length === 0) {
+    throw new common.NotFoundError('The specified pipeline coupling was not found.');
   }
-  return httph.ok_response(res, 
+  return httph.ok_response(res,
     JSON.stringify(pipeline_coupling_postgres_to_response(pipeline_coupling[0])));
 }
 
-const insert_pipeline_coupling = query.bind(query, fs.readFileSync('./sql/insert_pipeline_coupling.sql').toString('utf8'), (d) => { return d; });
-async function create_pipeline_coupling(pg_pool, app_uuid, app_name, space_name, org_uuid, pipeline, stage, required_status_checks) {
-  required_status_checks = required_status_checks || {"contexts":[]}
+const insert_pipeline_coupling = query.bind(query, fs.readFileSync('./sql/insert_pipeline_coupling.sql').toString('utf8'), (d) => d);
+async function create_pipeline_coupling(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Async function 'create_pipeline_coupling' has a complexity of 13. Maximum allowed is 8.](https://app.codacy.com/manual/Akkeris/controller-api/pullRequest?prid=5244943)